### PR TITLE
Adding support for HCP Boundary

### PR DESCRIPTION
**Pull request by rizkybiz**
### :hammer_and_wrench: Description
- Added support for creating an HCP Boundary cluster via Terraform
- Added docs for HCP Boundary resource and data source
- Updated hcp-sdk-go imports promoting consul to stable channel

### :ship: Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-hcp/blob/main/CHANGELOG.md):

```release-note
* provider: Added support for HCP Boundary Beta clusters
```

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [X] Have you added an acceptance test for the functionality being added?
- [X] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccBoundaryCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAccBoundaryCluster -timeout 210m
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/clients    0.495s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/consul     0.841s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/input      0.727s [no tests to run]
=== RUN   TestAccBoundaryCluster
--- PASS: TestAccBoundaryCluster (114.96s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   115.455s
```

### CODE REVIEW

File: docs/resources/boundary_cluster.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `cluster_id` (String) The ID of the Boundary cluster
+- `password` (String, Sensitive) The password of the initial admin user
```
Comment (xargs-P) [inline docs/resources/boundary_cluster.md:29]: 👋 Hello,
This is the first time any of the HCP resources are asking the user for a password in terraform. 🤔 
Could you add some type of notice to the user about how the input is stored in the state file or logs?
Similar to: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#password

**Review by xargs-P — CHANGES_REQUESTED** [review]

File: docs/resources/boundary_cluster.md (hunk 1 above)
Comment (rizkybiz) [reply docs/resources/boundary_cluster.md]: Good catch! I just made some changes to the resource and made note of how the password could show up in logs and is stored in the state file.

File: internal/provider/resource_boundary_cluster.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateStringNotEmpty,
+			},
+			"password": {
+				Description:      "The password of the initial admin user. Note that this may show up in logs, and it will be stored in the state file.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateStringNotEmpty,
```
Comment (lilic) [inline internal/provider/resource_boundary_cluster.go:62]: We have a few more validations that boundary enforces for this.

**Review by xargs-P — APPROVED** [review]
The password stuff is resolved ..

File: internal/provider/resource_boundary_cluster.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+			},
+			"password": {
+				Description:      "The password of the initial admin user. This must be at least 8 characters in length. Note that this may show up in logs, and it will be stored in the state file.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateBoundaryPassword,
+				Sensitive:        true,
+			},
+			"created_at": {
```
Comment (bcmdarroch) [inline internal/provider/resource_boundary_cluster.go:65]: ✨ Just to help with quick parsing of the inputs
```suggestion
// computed outputs
			"created_at": {
```

**Review by bcmdarroch — APPROVED** [review]
Thanks for the contribution! 🎉 Looks great to me!

RE: the dependency CI issue, I've done the bump in a separate PR as a work-around: https://github.com/hashicorp/terraform-provider-hcp/pull/381

File: internal/provider/resource_boundary_cluster.go [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+// defaultClusterTimeout is the amount of time that can elapse
+// before a cluster read operation should timeout.
+var defaultBoundaryClusterTimeout = time.Minute * 5
+
+// createUpdateBoundaryClusterTimeout is the amount of time that can elapse
+// before a cluster create operation should timeout.
+var createBoundaryClusterTimeout = time.Minute * 35
```
[the remark this answers is not in the document]
Comment (rizkybiz) [reply internal/provider/resource_boundary_cluster.go]: Bumped this down to 25 in a follow on commit. Thanks for the suggestion! Seems like HCP Boundary's architecture lends to relatively rapid creation and destruction.

File: internal/clients/client.go [hunk 5]
```diff
@@ -14,15 +14,18 @@ import (
 	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/preview/2019-12-10/client/project_service"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
-	cloud_consul "github.com/hashicorp/hcp-sdk-go/clients/cloud-consul-service/preview/2021-02-04/client"
-	"github.com/hashicorp/hcp-sdk-go/clients/cloud-consul-service/preview/2021-02-04/client/consul_service"
+	cloud_consul "github.com/hashicorp/hcp-sdk-go/clients/cloud-consul-service/stable/2021-02-04/client"
```
[the remark this answers is not in the document]
Comment (rizkybiz) [reply internal/clients/client.go:17]: No problem! Once I locked the hcp-sdk-go dep to the latest release version the imports were complaining so I figured I'd just bump it.

File: internal/provider/resource_boundary_cluster.go (hunk 3 above)
Comment (rizkybiz) [reply internal/provider/resource_boundary_cluster.go:66]: Updated! Good catch!

**Review by lilic — APPROVED** [review]
:shipit:

**Comment by bcmdarroch** [comment]
Quick test run after rebase shenanigans:
```
make testacc TESTARGS='-run=TestAccBoundaryCluster'

--- PASS: TestAccBoundaryCluster (146.49s)
PASS
```
🎉

**Pull request merged**